Engine/DataFeeds/TimeSlice.cs

```diff
@@ -145,13 +147,14 @@ public static TimeSlice Create(DateTime utcDateTime, DateTimeZone algorithmTimeZ
             var futuresChains = new FuturesChains(algorithmTime);
             var symbolChanges = new SymbolChangedEvents(algorithmTime);
 
-            foreach (var packet in data)
+            // ensure we read equity data before option data, so we can set the current underlying price
+            foreach (var packet in data.OrderBy(x => x.Configuration.Symbol.SecurityType))
```

Comment:
I wonder what impact this has on performance? Order by is historically slow since it has to enumerate the entire collection into a sorted collection and then re-enumerates the sorted collection

Reply:
I don't like it much either, but couldn't think of other simple ways to guarantee that equities will be listed before options. Perhaps forcing order of subscriptions for data feeds (not enough though, the method is also called for history requests) ?

Reply:
maybe we can benchmark it to see if it's worth addressing now or later?

Reply:
@StefanoRaggi -- were we able to get a benchmark on this? If it has little impact that I'd say we just merge it as is. Honestly, even if there is a noticeable impact we still may want to merge it and resolve the performance issue later -- I wonder what @jaredbroad thinks?

Reply:
@mchandschuh I ran backtests on a benchmark algorithm with 500 equities (3 runs on master + 3 runs on rebased branch) and didn't notice any backtesting speed reduction